Author: Christian Trenkwalder <christian@trenkwalder.it>
Description: switch from image.tostring() to image.tobytes()
 Traceback (most recent call last):
 [...]
   File "/usr/share/fofix/src/Texture.py", line 77, in loadImage
     string = image.tostring('raw', 'RGBA', 0, -1)
   File "/usr/lib/python2.7/dist-packages/PIL/Image.py", line 697, in tostring
     "Please call tobytes() instead.")
 NotImplementedError: tostring() has been removed. Please call tobytes() instead.

--- a/src/Texture.py
+++ b/src/Texture.py
@@ -74,19 +74,19 @@ class Texture:
     """Load the texture from a PIL image"""
     image = image.transpose(Image.FLIP_TOP_BOTTOM)
     if image.mode == "RGBA":
-      string = image.tostring('raw', 'RGBA', 0, -1)
+      string = image.tobytes('raw', 'RGBA', 0, -1)
       self.loadRaw(image.size, string, GL_RGBA, 4)
     elif image.mode == "RGB":
-      string = image.tostring('raw', 'RGB', 0, -1)
+      string = image.tobytes('raw', 'RGB', 0, -1)
       self.loadRaw(image.size, string, GL_RGB, 3)
     elif image.mode == "L":
-      string = image.tostring('raw', 'L', 0, -1)
+      string = image.tobytes('raw', 'L', 0, -1)
       self.loadRaw(image.size, string, GL_LUMINANCE, 1)
     else:
       try:
         image = image.convert('RGB')
         Log.warn("Unsupported image mode '%s' converted to 'RGB'. May have unexpected results." % image.mode)
-        string = image.tostring('raw', 'RGB', 0, -1)
+        string = image.tobytes('raw', 'RGB', 0, -1)
         self.loadRaw(image.size, string, GL_RGB, 3)
       except:
         raise TextureException("Unsupported image mode '%s'" % image.mode)
@@ -113,7 +113,7 @@ class Texture:
       # appears to be using PIL to do the conversion.
       string = pygame.image.tostring(surface, "RGB")
       image = Image.fromstring("RGB", surface.get_size(), string).convert("L")
-      string = image.tostring('raw', 'L', 0, -1)
+      string = image.tobytes('raw', 'L', 0, -1)
       self.loadRaw(surface.get_size(), string, GL_LUMINANCE, GL_INTENSITY8)
     else:
       if alphaChannel:
@@ -132,7 +132,7 @@ class Texture:
       # appears to be using PIL to do the conversion.
       string = pygame.image.tostring(surface, "RGB")
       image = Image.fromstring("RGB", surface.get_size(), string).convert("L")
-      string = image.tostring('raw', 'L', 0, -1)
+      string = image.tobytes('raw', 'L', 0, -1)
       self.loadSubRaw(surface.get_size(), position, string, GL_INTENSITY8)
     else:
       if alphaChannel:
